debezium-connector-mysql/src/main/java/io/debezium/connector/mysql/MySqlReadOnlyIncrementalSnapshotChangeEventSource.java

```diff
@@ -97,25 +97,21 @@ public MySqlReadOnlyIncrementalSnapshotChangeEventSource(RelationalDatabaseConne
     public void init(MySqlPartition partition, OffsetContext offsetContext) {
         super.init(partition, offsetContext);
 
-        if (isKafkaChannelEnabled()) {
-
-            restoreOffset();
-        }
+        restoreOffset();
     }
 
     private void restoreOffset() {
-
+        // Restore Kafka signal offsets if needed
         KafkaSignalChannel kafkaSignal = dispatcher.getSignalProcessor().getSignalChannel(KafkaSignalChannel.class);
+        if (kafkaSignal == null) {
```

Comment:
This code is not equivalent. The`getSignalChannel` method only return registered channels but channel can be not enabled by user configuration.

Reply:
@mfvitale Thanks for explanation. Could you please remove my commit and just add a JavaDoc to the method? Then feel free to merge yourself.